redash/monitor.py

```diff
@@ -18,29 +23,20 @@ def get_object_counts():
 
 
 def get_queues():
-    queues = {}
-    for ds in models.DataSource.query:
-        for queue in (ds.queue_name, ds.scheduled_queue_name):
-            queues.setdefault(queue, set())
-            queues[queue].add(ds.name)
+    query = models.db.session.execute(
+        'select distinct queue_name from data_sources union all select distinct scheduled_queue_name from data_sources')
 
-    return queues
+    return ['celery'] + [row[0] for row in query]
 
 
 def get_queues_status():
-    queues = get_queues()
+    queues = {}
```

Comment:
```suggestion
    return dict((q, {'size': redis_connection.llen(q)}) for q in get_queues())
```

Reply:
I'll pass on this one. Feels less readable to me: even knowing what it should do, I got confused for a moment about what exactly it does.